invokeai/backend/model_manager/load/load_default.py

```diff
@@ -0,0 +1,188 @@
+        self._ram_cache.make_room(self.get_size_fs(config, model_path, submodel_type))
+
+        # This is where the model is actually loaded!
+        with skip_torch_weight_init():
+            loaded_model = self._load_model(model_path, model_variant=model_variant, submodel_type=submodel_type)
+
+        self._ram_cache.put(
+            config.key,
+            submodel_type=submodel_type,
+            model=loaded_model,
+            size=calc_model_size_by_data(loaded_model),
+        )
+
+        return self._ram_cache.get(
+            key=config.key,
+            submodel_type=submodel_type,
+            stats_name=":".join([config.base, config.type, config.name, (submodel_type or "")]),
+        )
+
+    def get_size_fs(
+        self, config: AnyModelConfig, model_path: Path, submodel_type: Optional[SubModelType] = None
+    ) -> int:
+        """Get the size of the model on disk."""
+        return calc_model_size_by_fs(
+            model_path=model_path,
+            subfolder=submodel_type.value if submodel_type else None,
+            variant=config.repo_variant if hasattr(config, "repo_variant") else None,
+        )
+
+    def _load_diffusers_config(self, model_path: Path, config_name: str = "config.json") -> Dict[str, Any]:
+        return ConfigLoader.load_config(model_path, config_name=config_name)
+
+    # TO DO: Add exception handling
+    def _hf_definition_to_type(self, module: str, class_name: str) -> ModelMixin:  # fix with correct type
+        if module in ["diffusers", "transformers"]:
+            res_type = sys.modules[module]
+        else:
+            res_type = sys.modules["diffusers"].pipelines
+        result: ModelMixin = getattr(res_type, class_name)
+        return result
+
+    # TO DO: Add exception handling
+    def _get_hf_load_class(self, model_path: Path, submodel_type: Optional[SubModelType] = None) -> ModelMixin:
+        if submodel_type:
+            try:
+                config = self._load_diffusers_config(model_path, config_name="model_index.json")
+                module, class_name = config[submodel_type.value]
+                return self._hf_definition_to_type(module=module, class_name=class_name)
+            except KeyError as e:
+                raise InvalidModelConfigException(
+                    f'The "{submodel_type}" submodel is not available for this model.'
+                ) from e
+        else:
+            try:
+                config = self._load_diffusers_config(model_path, config_name="config.json")
+                class_name = config.get("_class_name", None)
+                if class_name:
+                    return self._hf_definition_to_type(module="diffusers", class_name=class_name)
+                if config.get("model_type", None) == "clip_vision_model":
+                    class_name = config.get("architectures")[0]
+                    return self._hf_definition_to_type(module="transformers", class_name=class_name)
```

Comment:
This feels like something that should be handled in a `ModelLoader` sub-class where we know what model we are loading, rather than pushing it up to the base class.

It makes me wonder if `_get_hf_load_class(...)` should exist at all. It seems harder to keep this working for a bunch of model types than to just have each model handle for itself.

Reply:
This method and its dependencies have been moved from the base class into the `ModelLoader` subclass named `GenericDiffusersLoader`. This is the parent of the `StableDiffusionLoader` and `OnnxLoader` subclasses . As long as the model in question is in diffusers format, `_get_hf_load_class()` will work and will be easier to maintain than us trying to keep a table of which diffusers classes to use to load a model.